### DISCUSSION TITLE
Bug: Duplicate typing

### DISCUSSION THREAD

**Discussion by GitHubIL** [AppGround-io/bluetooth-keyboard-and-mouse-support#197] [Bugs & Problems]
Great app.
There is a small bug, when I type on the phone for the streaming device, every time I make space after the first word, the first word is doubled. Duplication occurs only on the first word.
Another thing, when the phone and the streaming device are both in English mode, when I press a dot on the phone it comes out in the streaming device as a slash.

**Comment by AppGround-io** [comment]
This was reported was reported to me before, but I was not yet able to reproduce it on any of my devices.

Please let me know which device, Android version and the keyboard app you are using so that I can try to reproduce it on my side.

**Comment by Staini** [comment]
I can confirm it, too.
It happens only if I use the Samsung Keyboard (5.1.30.0). If I use an other keyboard it doesn't happen.

Device: Galaxy Note 9 with Android 10.

**Reply by AppGround-io** [reply]
@Staini Thank you for confirming the problem. 
I tried to reproduce it on a Galaxy Tab S4 but the problem did not happen. 
Would it be possible for you to take a video of the app when this problem happens?

**Reply by Staini** [reply]
I think I have found the problem. It's the Samsung "auto spell check". If it's disabled, duplicate words will no longer occur.

Here is an example with auto spell check:
test **[space key]**
_writes:_ test 

tesd **[space key]**
_writes:_ tesdtesd

(If you still need a video, I csn gladly make one)

**Reply by GitHubIL** [reply]
Cool, I checked now, after canceling the auto-spell check, there is no problem of duplication.

**Reply by GitHubIL** [reply]
interesting. When I write in another language (Hebrew), there is no duplication - even though automatic spell checking works.

**Reply by alexanderadam** [reply]
I had the same issue a few times. Also on a Samsung device (I reproduced this issue on Galaxy Tab S4, S6 and S7).

> Another thing, when the phone and the streaming device are both in English mode, when I press a dot on the phone it comes out in the streaming device as a slash.

The mismapping is most likely a duplicate of the [issues 189](https://github.com/AppGround-io/bluetooth-keyboard-and-mouse-support/discussions/189) and [216](https://github.com/AppGround-io/bluetooth-keyboard-and-mouse-support/discussions/216) (actually is a "discussion" but it would be an issue in regular projects obviously)

**Reply by GitHubIL** [reply]
Hi
I have already written [here ](https://github.com/AppGround-io/bluetooth-keyboard-and-mouse-support/discussions/197#discussioncomment-2120840)that the problem is because I did not change the language in the app.
when using two languages, every time you change a language on the phone you also have to change the language in the app itself:
![Bluetooth Keyboard   Mouse](https://user-images.githubusercontent.com/67213314/152701998-c2c5be4d-804b-4a5d-b7f4-aa7122624505.jpg)

**Comment by GitHubIL** [comment]
I installed the latest version of the app, free version 4.9.2.
My device is a Galaxy s7 with Android 10.

As [@Staini](https://github.com/Staini) wrote, with the Google Keyboard app there is no problem. But when I use my default keyboard app - Samsung Keyboard version 5.2.10.25 - there is a problem of duplication after space.
There is also another problem - the Enter key does not work, even with the Google Keyboard app!
Besides, a great app.

As for the first bug I wrote (that a dot key comes out as a slash), the problem is with me because I use two languages, and I have not changed the language through the app before. Anyway it would be nice if changing language through the keyboard app on the phone - will automatically change the language in the remote app (because now to change language you need three actions: 1.Keyboard app. 2.Remote app. 3.Change language on the streaming device).

Also, if it is possible to add these things:
- Bottom scroll bar.
- Change the volume even when the screen is off.
- Option to cancel brightness change through the app.
- Option to set the app to be in Portrait mode only (Disables auto-screen rotation only for this app).

Thank you

**Comment by GitHubIL** [comment]
updating:
1. As @Staini wrote [here ](https://github.com/AppGround-io/bluetooth-keyboard-and-mouse-support/discussions/197#discussioncomment-2120954)- The problem of duplication in typing occurs in Samsung's keyboard app when automatic spell checking is enabled.
Anyway, for some reason this problem does not exist when I change to another language (Hebrew) even though the automatic spell check is enabled.

2. There is another bug, which Enter key does not work, even in Google keyboard app.
Does not work I mean, for example to make a new line in the text, or in the address bar of the browser. But for example going into a folder it works.

**Comment by AppGround-io** [comment]
@Staini Thank you very much for investigating the issue, I was able to reproduce the problem now and I will provide a fix for it in the next version.

@GitHubIL Thank you for the feature requests. It would be great if you could create separate tickets for each request so that is doesn't get lost and it will be easier for others to comment and the specific request.

**Comment by GitHubIL** [comment]
What about the bug that Enter key does not work. Is also for him to create a separate ticket, or just for these four things:
> * Bottom scroll bar.
> * Change the volume even when the screen is off.
> * Option to cancel brightness change through the app.
> * Option to set the app to be in Portrait mode only (Disables auto-screen rotation only for this app).

Thank you
